_posts/2016-03-02-Rust-1.7.md

```diff
@@ -0,0 +1,194 @@
+---
+layout: post
+title: "Announcing Rust 1.7"
+author: The Rust Core Team
+---
+
+The Rust team is happy to announce the latest version of Rust, 1.7. Rust is a
+systems programming language focused on safety, speed, and concurrency.
+
+As always, you can [install Rust 1.7][install] from the appropriate page on our
+website, and check out the [detailed release notes for 1.7][notes] on GitHub.
+About 1300 patches were landed in this release.
+
+[install]: http://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/stable/RELEASES.md#version-170-2016-03-03
+
+### What's in 1.7 stable
+
+This release is primarily about library features. While we have several
+language features cooking for future releases, the timeframe in which 1.7
+included the holidays, which means less time for commenting on GitHub and more
+time for spending with loved ones.
+
+#### New Language features
+
+`None`
+
+#### Library stabilizations
+
+About 40 library functions and methods are now stable in 1.7. Notable
+improvements include:
+
+* The ability to customize `HashMap<K, V>`s and similar data structures with
```

Comment:
Sure yeah, perhaps something like (wordsmithing always welcome):

---

One of the largest APIs stabilized during the 1.7 cycle was support for custom hash algorithms in the standard library's `HashMap` type. Previously all hash maps would use [SipHash](https://en.wikipedia.org/wiki/SipHash) as the hashing algorithm, which provides protection against DOS attacks by default. SipHash, however, is [not very fast](http://cglab.ca/~abeinges/blah/hash-rs/) at hashing small keys. As shown, however, the [FNV hash algorithm](https://en.wikipedia.org/wiki/Fowler%E2%80%93Noll%E2%80%93Vo_hash_function) is much faster for these size of inputs. This means that by switching hash algorithms for types like `HashMap<usize, V>` there can be a significant speedup so long as the loss of DOS protection is acceptable. 

To see this in action, you can check out the [fnv crate](https://crates.io/crates/fnv) on crates.io and create a `HashMap` via:

```rust
extern crate fnv;

use std::collections::HashMap;                            
use std::hash::BuildHasherDefault;                        
use fnv::FnvHasher;                                       
                                                          
type MyHasher = BuildHasherDefault<FnvHasher>;            
                                                          
let mut map: HashMap<_, _, MyHasher> = HashMap::default();
map.insert(1, "Hello");                                   
map.insert(2, ", world!");                                
println!("{:?}", map);
```

Note that most of the time you don't even need to specify the hasher as type inference will take care of it, so `HashMap::default()` should be all you need to get up to 2x faster hashes. It's also worth pointing out that [`Hash` trait](http://doc.rust-lang.org/std/hash/trait.Hash.html) is agnostic to the hashing algorithm used, so no changes are needed to the types being inserted into hash maps to reap the benefits!